frameworks/hdfs/integration/tests/test_sanity.py

```diff
@@ -55,6 +55,89 @@ def test_bump_journal_cpus():
 
 
 @pytest.mark.sanity
+def test_modify_app_config_update():
+    check_health()
+    app_config_field = 'TASKCFG_ALL_CLIENT_READ_SHORTCIRCUIT_STREAMS_CACHE_SIZE_EXPIRY_MS'
+
+    journal_ids = get_task_ids('journal')
+    name_ids = get_task_ids('name')
+    zkfc_ids = get_task_ids('zkfc')
+    data_ids = get_task_ids('data')
+    print('journal ids: ' + str(journal_ids))
+    print('name ids: ' + str(name_ids))
+    print('zkfc ids: ' + str(zkfc_ids))
+    print('data ids: ' + str(data_ids))
+
+    config = get_marathon_config()
+    print('marathon config: ')
+    print(config)
+    expiry_ms = int(config['env'][app_config_field])
+    config['env'][app_config_field] = str(expiry_ms + 1)
+    r = request(
+        dcos.http.put,
+        marathon_api_url('apps/' + PACKAGE_NAME),
+        json=config)
+
+    # All tasks should be updated because hdfs-site.xml has changed
+    tasks_updated('journal', journal_ids)
+    tasks_updated('name', name_ids)
+    tasks_updated('zkfc', zkfc_ids)
+    tasks_updated('data', data_ids)
+
+    check_health()
+
+
+@pytest.mark.sanity
+def test_modify_app_config_rollback():
+    check_health()
+    app_config_field = 'TASKCFG_ALL_CLIENT_READ_SHORTCIRCUIT_STREAMS_CACHE_SIZE_EXPIRY_MS'
+
+    journal_ids = get_task_ids('journal')
+    name_ids = get_task_ids('name')
+    zkfc_ids = get_task_ids('zkfc')
+    data_ids = get_task_ids('data')
+    print('journal ids: ' + str(journal_ids))
+    print('name ids: ' + str(name_ids))
+    print('zkfc ids: ' + str(zkfc_ids))
+    print('data ids: ' + str(data_ids))
+
+    old_config = get_marathon_config()
+    config = get_marathon_config()
+    print('marathon config: ')
+    print(config)
+    expiry_ms = int(config['env'][app_config_field])
+    print('expiry ms: ' + str(expiry_ms))
+    config['env'][app_config_field] = str(expiry_ms + 1)
+    r = request(
+        dcos.http.put,
+        marathon_api_url('apps/' + PACKAGE_NAME),
+        json=config)
+
+    # Wait for journal nodes to be affected by the change
+    tasks_updated('journal', journal_ids)
+    journal_ids = get_task_ids('journal')
+
+    print('old config: ')
+    print(old_config)
+    # Put the old config back (rollback)
```

Comment:
Is there a chance that this could be flaky, if the zkfc nodes begin to apply config updates before the rollback request completes?